src/Quadrature.jl

```diff
@@ -372,6 +390,66 @@ function __init__()
     end
 end
 
+ZygoteRules.@adjoint function __solvebp(prob,alg,sensealg,lb,ub,p,args...;kwargs...)
+    out = __solvebp(prob,alg,sensealg,lb,ub,p,args...;kwargs...)
+    function quadrature_adjoint(Δ)
+        y = typeof(Δ) <: Array{<:Number,0} ? Δ[1] : Δ
+        if isinplace(prob)
+            if lb isa Number && prob.batch == 0
+                dx = Float64[lb]
+            elseif lb isa Number
+                dx = zeros(length(lb),prob.batch)
+            elseif prob.batch == 0
+                dx = zeros(length(lb))
+            else
+                dx = zeros(length(lb),prob.batch)
+            end
+            _f = (x) -> (prob.f(dx,x,p); dx)
+
+            if sensealg.vjp isa ZygoteVJP
+                dfdp = function (dx,x,p)
+                    _,back = Zygote.pullback(p) do p
+                        dx = Zygote.Buffer(x)
+                        prob.f(dx,x,p)
+                        copy(dx)
+                    end
+                    back(y)[1]
+                end
+            elseif sensealg.vjp isa ReverseDiffVJP
+                error("TODO")
+            end
+        else
+            _f = (x) -> prob.f(x,p)
+            if sensealg.vjp isa ZygoteVJP
+                dfdp = function (x,p)
+                    _,back = Zygote.pullback(p->prob.f(x,p),p)
+                    back(y)[1]
+                end
+            elseif sensealg.vjp isa ReverseDiffVJP
+                error("TODO")
+            end
+        end
+        dlb = -_f(lb)
+        dub = _f(ub)
+
+        dp_prob = QuadratureProblem(dfdp,lb,ub,p;nout=length(p),batch = prob.batch,kwargs...)
```

Comment:
It seems that the boundary term is missing. Will Zygote handle parameter-dependent `lb` and `ub` automatically?

Reply:
yes, as long as this definition for the derivative at the boundary is correct. So for 1-D this already works. For N-D I need to think more (Stokes Theorem, etc.)

Reply:
I set it to return `nothing` in the cases it's incorrect for, for now. Our applications don't need this derivative anyways.